### ISSUE TITLE
Could you provide such SP regularization on Pytorch?

### ISSUE THREAD

**Issue by PkuRainBow** [holyseven/TransferLearningClassification#1]
really great work!

I am wondering could you share us the pytorch version?

**Comment by holyseven** [comment]
em... just for information, my implement of L2-SP in tf can be found [here](https://github.com/holyseven/TransferLearningClassification/blob/master/model/network_base.py#L138-L156). And I am not working on pytorch at this moment, but I think there is no large difference with tf. 

See the discussion of [how to manually implement L1 or L2 in pytorch](https://discuss.pytorch.org/t/how-does-one-implement-weight-regularization-l1-or-l2-manually-without-optimum/7951) and the extra need to do is:
load the pre-trained model, find the corresponding parameters and do the L2-SP.

**Comment by holyseven** [comment]
See if this is interesting for you. https://github.com/holyseven/PSPNet-TF-Reproduce/issues/1#issuecomment-396004749

If you could try L2-SP (and other SP regularizers) with experiments in Pytorch (classification or segmentation, or other transfer learning applications), thanks in advance for sharing.

**Comment by PkuRainBow** [comment]
@holyseven 
We have implemented the original L2 norm regularizer, but the performance with our implementation fails to match the performance of the default L2 norm regularizer.

I am wondering whether have you considered which parameters to add the L2 norm regularizer, maybe the BN layer does not need L2 norm regularizer.

**Comment by holyseven** [comment]
I was told that Facebook regularized biases too. And I checked the code of [torch.optim.SGD](https://github.com/pytorch/pytorch/blob/master/torch/optim/sgd.py#L93) for example, by default, it computes the weight decay for **all** parameters. 

Just for info, I made a mistake and I updated the code from the link I previously proposed, changing `w.norm(2)` to `torch.pow(w, 2).sum()`.
